scripts/g.extension/g.extension.py

```diff
@@ -2299,46 +2351,45 @@ def resolve_source_code(url=None, name=None, branch=None, fork=False):
     >>> resolve_source_code('https://bitbucket.org/joe-user/grass-module') # doctest: +SKIP
     ('remote_zip', 'https://bitbucket.org/joe-user/grass-module/get/default.zip')
     """
+    # Handle URL for the offical repo
     if not url and name:
         module_class = get_module_class_name(name)
-        # note: 'trunk' is required to make URL usable for 'svn export' call
+        repo_branches = get_github_branches()
+        # Define branch to fetch from (latest or current version)
+        version_branch = "grass{}".format(version[0])
+        if version_branch not in repo_branches:
+            version_branch = repo_branches[-1]
```

Comment:
Given that the API has rate limits, you can consider abandoning the idea of this ideal test and go with something like:

```suggestion
        # Hard-coded value which needs to be updated when new official addon branch is created.
        addons_available_up_to_version = 7  # perhaps as a global ADDONS_AVAILABLE...
        use_version = int(version[0])
        if use_version > addons_available_up_to_version:
            # Fallback to the highest available version.
            use_version = addons_available_up_to_version
        version_branch = f"grass{use_version}"
```

Reply:
Yes, agreed. Usage of github API is abandoned for now. The rate limit for un-authenticated API calls in [github](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#increasing-the-unauthenticated-rate-limit-for-oauth-applications) seems to be 60/h (comaprd to [gitlab](https://docs.gitlab.com/ee/security/rate_limits.html) with a least 10 times as many).
As the fallback solution I went for fetching from version branch if no branch is given and from default branch, if the version branch does not exist. Maybe we should add in the documentation, that branches in forks from the official repo have to follow the structure of the default branch there (with `src` directory at the root)...

Reply:
Agreed that documentaion is needed for that, but maybe in another PR. The whole "fork" think here is suboptimal. Typical (soft) fork on GitHub will have the branches. Do we mean hard project fork? I think the aim is institutional repositories (private or public), right? That was the case in the past, I think. I'm not sure how to call "a repo with the same structure as the official repo", plus there is the issues with the generated metadata file which is separate if I recall correctly. ...next PR. Perhaps open an issue?